### PR TITLE
Sticky ad: tidy

### PR DESCRIPTION
**Pull request by OliverJAsh**
Comments and tidy up, see commits for breakdown.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/navigation/newSticky.js [hunk 1]
```diff
@@ -107,48 +113,36 @@ define([
...
-            'transition': ''
-        });
-
-        // Set
         // TODO: move into stylesheets when productionised
         els.$document.addClass('new-sticky-ad');
-        els.$header.css({ 'margin-top': state.adHeight });
-        els.$adBanner.css({ 'max-height': state.adHeight });
+
+        var transitionTimingFunction = 'cubic-bezier(0, 0, 0, .985)';
```
Comment (jamesgorrie) [inline static/src/javascripts/projects/common/modules/navigation/newSticky.js]: For my info, `.985`?

File: static/src/javascripts/projects/common/modules/navigation/newSticky.js [hunk 2]
```diff
@@ -107,48 +113,36 @@ define([
...
+        var transitionDuration = '1s';
+
+        els.$header.css({
+            'transition': state.shouldTransition
```
Comment (jamesgorrie) [inline static/src/javascripts/projects/common/modules/navigation/newSticky.js]: Would this not set the transition to an invalid CSS value?
Perhaps just not setting it if `!shouldTransion`.

File: static/src/javascripts/projects/common/modules/navigation/newSticky.js [hunk 3]
```diff
@@ -107,48 +113,36 @@ define([
...
+
+        els.$adBanner.css({
+            'position': userHasScrolledPastHeader ? 'absolute' : 'fixed',
+            'top': userHasScrolledPastHeader ? state.headerHeight : '',
+            'height': state.adHeight,
+            // Stop the ad from overflowing while we transition
+            'overflow': state.shouldTransition ? 'hidden' : '',
+            'transition': state.shouldTransition
+                ? ['height', transitionDuration, transitionTimingFunction].join(' ')
+                : ''
```
Comment (jamesgorrie) [inline static/src/javascripts/projects/common/modules/navigation/newSticky.js]: [See here](https://github.com/guardian/frontend/pull/11916/files#r52463403), and line 137?

File: static/src/javascripts/projects/common/modules/navigation/newSticky.js (hunk 1 above)
Comment (OliverJAsh) [inline static/src/javascripts/projects/common/modules/navigation/newSticky.js]: We've just copied this from what we had with the old sticky nav: https://github.com/guardian/frontend/blob/master/static/src/stylesheets/module/_adslot.scss#L120

It looks like this: 
image

@stephanfowler and I did think about changing it to ease-in

File: static/src/javascripts/projects/common/modules/navigation/newSticky.js (hunk 3 above)
Comment (OliverJAsh) [inline static/src/javascripts/projects/common/modules/navigation/newSticky.js]: Setting it to `''` actually unsets the inline style

File: static/src/javascripts/projects/common/modules/navigation/newSticky.js (hunk 2 above)
Comment (OliverJAsh) [inline static/src/javascripts/projects/common/modules/navigation/newSticky.js]: Setting it to `''` actually unsets the inline style. I used to do what you're saying (`only set it if `shouldTransition`) but this way I can use a single object to set all styles for a given element. Not sure if the `''` to unset is clear though

File: static/src/javascripts/projects/common/modules/navigation/newSticky.js [hunk 4]
```diff
@@ -179,41 +173,39 @@ define([
         });
     };
 
+    var reducer = function (previousState, action) {
+        switch (action.type) {
+            case 'SCROLL':
+                return assign({}, previousState, {
+                    previousAdHeight: previousState.adHeight,
```
Comment (jamesgorrie) [inline static/src/javascripts/projects/common/modules/navigation/newSticky.js]: As you're assigning, this isn't necessary is it?

Comment (jamesgorrie) [inline static/src/javascripts/projects/common/modules/navigation/newSticky.js]: Oh wait, just reread it.

File: static/src/javascripts/projects/common/modules/navigation/newSticky.js (hunk 2 above)
Comment (jamesgorrie) [inline static/src/javascripts/projects/common/modules/navigation/newSticky.js]: If it clears it, all good :+1:

**Comment by jamesgorrie** [comment]
Looks good form here :+1:

File: static/src/javascripts/projects/common/modules/navigation/newSticky.js (hunk 2 above)
Comment (sndrs) [inline static/src/javascripts/projects/common/modules/navigation/newSticky.js]: `''` or `null` should clear it

**Comment by Calanthe** [comment]
@OliverJAsh  @Jholder112233 I just realised that after introducing the sticky nav, we had some bugs with anchor links which were partially fixed: here https://github.com/guardian/frontend/pull/9986 and here https://github.com/guardian/frontend/pull/9867. Now when the sticky nav is gone, we should double check and revert those fixes.

**Comment by OliverJAsh** [comment]
Thanks @Calanthe!

**Pull request merged**